### PR TITLE
Feature/use wikipedia api

### PR DESCRIPTION
**Pull request by zsarge**
# Use the Wikipedia API

**TL;DR:** This PR enables:
✅ pulling from the Wikipedia API for page content
✅ filtering page content via CSS instead of regex
  - I was initally against hiding content with CSS because I wanted to limit the amount of potentially hackable material that we were shipping to the client. However, given that the entire page could be modified by the client anyway, I think it's okay to ship them some content and then hide it via CSS. This ensures proper DOM modeling, which we did not ensure when parsing with regex. 
  
✅ I have confirmed this supports mobile

# TODO AFTER MERGING

Use @ConGodsted's `current year` feature to embed "TechOlympics ${Current Year}" into the page template.

## Using the Wikipedia API?

- There are many NodeJS Wikipedia APIs
  1. [Wikipedia](https://github.com/dopecodez/Wikipedia)
    - returns images in a nice format
    - looks like it's already in use by <https://wiki-race.com/game>
  2. [NodeMW](https://github.com/macbre/nodemw)
  3. [NodeJS wiki](https://github.com/dijs/wiki)
- Excluding references from parse request?
  - https://stackoverflow.com/a/16266061
  - https://stackoverflow.com/a/5922089 
- Problem:
  - How do we avoid links that do not point to wikipedia?
  - ([Thou shalt not parse HTML with regex](https://stackoverflow.com/a/1732453))
  - I think I'm still going to parse HTML with regex. I can't think of a more performant solution.
  
When I started this, I misremembered Wikipedia's API to be more cusomizable than it really is.

In reality, it looks like we can reduce the amount of parsing we're doing by not getting the entire full page content. I believe this is also a better decision in terms of long-term mainence, because it reduces the amount that could change, as the API should be way more stable. However, I don't believe we're going to be able to eliminate the parsing step. Just to be clear. 

Let me know if anyone sees anything that I'm missing.

# How do other people do it?

I'm writing this out as an exercise to see how other people have solved this problem.

## Default Wikipedia
Screenshot from 2023-12-18 14-29-52
## The Wiki Game
Screenshot from 2023-12-18 14-30-24
They seem to either be using a predownloaded archive, or to be using a long cache duration, as it seems that their page is well behind the current page.

| Feature                         | Notes    |
|---------------------------------|----------|
| References                  | Removed  |
| Embedded elements such as music | Removed  |
| External links                  | Removed  |
| Table design                    | Degraded |
| Bullet points                   | Degraded |

## [Wiki-Race.com](https://wiki-race.com/game)
Screenshot from 2023-12-18 14-32-31

Their parsing step adds data to each link, which is then used to redirect on click:

```html
<a href="https://en.wikipedia.org/wiki/Saxophone" data-wiki-page="Saxophone">saxophone</a>
```

| Feature                         | Notes    |
|---------------------------------|----------|
| References                  | Included with links removed  |
| Embedded elements such as music | Supported  |
| External links                  | Removed  |
| Table design                    | Supported |
| Bullet points                   | Supported |

:star: Whatever they're doing has good page quality.

## [The Wikipedia Game](https://www.thewikipediagame.com/)

Screenshot from 2023-12-18 14-31-31

| Feature                         | Notes    |
|---------------------------------|----------|
| References                  | Included with links removed  |
| Embedded elements such as music | Degraded  |
| External links                  | Removed  |
| Table design                    | Supported |
| Bullet points                   | Supported |
## Wiki Races Classic
Screenshot from 2023-12-18 14-30-34

## This PR
image

### CODE REVIEW

**Comment by zsarge** [comment]
I'd be happy to resolve the merge conflicts by taking the older versions of levels.json. Or we can just ignore them.

**Comment by cassiancc** [comment]
Great work! Going ahead and merging.
